src/arrayMethodJoin.js

```diff
@@ -1,11 +1,41 @@
 'use strict';
 
-/**
+/*
  * Implement method join
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+
+    let newStr = `${this[0]}`;
+
+    if (this[0] === undefined || this[0] === null) {
+      newStr = '';
+    }
+
+    let separatorDefault = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separatorDefault === null) {
+      separatorDefault = 'null';
+    }
+
+    if (separatorDefault === undefined) {
+      separatorDefault = ',';
+    }
```

Comment:
We don't have a condition to check separator by "Null", only by "Undefined", so you can use value by default.

Reply:
thank, very good idea)